File: frontend/lib/types/data-bag.ts
export interface DataBag {
  id: string
  projectId: string
  name: string
  description?: string
  dataSchema?: Record<string, any>
  createdAt: string
  updatedAt: string
}

export interface DataBagItem {
  id: string
  dataBagId: string
  data: Record<string, any>
  createdAt: string
}

export interface DataBagWithItems {
  dataBag: DataBag
  items: DataBagItem[]
}

export interface RequirementDataBagLink {
  id: string
  requirementId: string
  dataBagId: string
  dataBagItemId?: string
  createdAt: string
}

export interface CreateDataBagRequest {
  name: string
  description?: string
  dataSchema?: Record<string, any>
}

export interface UpdateDataBagRequest {
  name?: string
  description?: string
  dataSchema?: Record<string, any>
}

export interface ImportDataRequest {
  items: Record<string, any>[]
}

export interface LinkDataBagRequest {
  dataBagId: string
  dataBagItemId?: string
}

File: backend/src/models/data_bag.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Serialize, Deserialize, FromRow)]
#[serde(rename_all = "camelCase")]
pub struct DataBag {
    pub id: Uuid,
    pub project_id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub data_schema: Option<serde_json::Value>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
#[serde(rename_all = "camelCase")]
pub struct DataBagItem {
    pub id: Uuid,
    pub data_bag_id: Uuid,
    pub data: serde_json::Value,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
#[serde(rename_all = "camelCase")]
pub struct RequirementDataBagLink {
    pub id: Uuid,
    pub requirement_id: Uuid,
    pub data_bag_id: Uuid,
    pub data_bag_item_id: Option<Uuid>,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DataBagWithItems {
    pub data_bag: DataBag,
    pub items: Vec<DataBagItem>,
}
